Compiling with Kokkos development branch doesn't work.

Error messages:
```
In file included from Arborx/src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp:17:0,
                 from Arborx/test/tstDetailsDistributedSearchTreeImpl.cpp:12:
Arborx/src/details/ArborX_DetailsUtils.hpp: In function ‘std::pair<typename ViewType::non_const_value_type, typename ViewType::non_const_value_type> ArborX::minMax(const ViewType&)’:
Arborx/src/details/ArborX_DetailsUtils.hpp:176:27: error: ‘MinMaxScalar’ is not a member of ‘Kokkos::Experimental’
     Kokkos::Experimental::MinMaxScalar<typename ViewType::non_const_value_type>
                           ^~~~~~~~~~~~
Arborx/src/details/ArborX_DetailsUtils.hpp:176:27: note: suggested alternative:
In file included from kokkos-install/include/Kokkos_Parallel.hpp:255:0,
                 from kokkos-install/include/Kokkos_Serial.hpp:55,
                 from kokkos-install/include/Kokkos_Core.hpp:53,
                 from Arborx/src/details/ArborX_DetailsDistributor.hpp:16,
                 from Arborx/src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp:14,
                 from Arborx/test/tstDetailsDistributedSearchTreeImpl.cpp:12:
kokkos-install/include/Kokkos_Parallel_Reduce.hpp:633:8: note:   ‘Kokkos::MinMaxScalar’
 struct MinMaxScalar {
        ^~~~~~~~~~~~
In file included from Arborx/src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp:17:0,
                 from Arborx/test/tstDetailsDistributedSearchTreeImpl.cpp:12:
Arborx/src/details/ArborX_DetailsUtils.hpp:176:79: error: expected ‘(’ before ‘>’ token
     Kokkos::Experimental::MinMaxScalar<typename ViewType::non_const_value_type>
                                                                               ^
Arborx/src/details/ArborX_DetailsUtils.hpp:177:9: error: ‘result’ was not declared in this scope
         result;
         ^~~~~~
Arborx/src/details/ArborX_DetailsUtils.hpp:178:27: error: ‘MinMax’ is not a member of ‘Kokkos::Experimental’
     Kokkos::Experimental::MinMax<typename ViewType::non_const_value_type>
                           ^~~~~~
Arborx/src/details/ArborX_DetailsUtils.hpp:178:27: note: suggested alternative:
In file included from kokkos-install/include/Kokkos_Core.hpp:50:0,
                 from Arborx/src/details/ArborX_DetailsDistributor.hpp:16,
                 from Arborx/src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp:14,
                 from Arborx/test/tstDetailsDistributedSearchTreeImpl.cpp:12:
kokkos-install/include/Kokkos_Core_fwd.hpp:317:61: note:   ‘Kokkos::MinMax’
 template<class ScalarType , class Space = HostSpace> struct MinMax;
                                                             ^~~~~~
In file included from Arborx/src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp:17:0,
                 from Arborx/test/tstDetailsDistributedSearchTreeImpl.cpp:12:
Arborx/src/details/ArborX_DetailsUtils.hpp:178:73: error: expected ‘(’ before ‘>’ token
     Kokkos::Experimental::MinMax<typename ViewType::non_const_value_type>
                                                                         ^
Arborx/src/details/ArborX_DetailsUtils.hpp:179:9: error: there are no arguments to ‘reducer’ that depend on a template parameter, so a declaration of ‘reducer’ must be available [-fpermissive]
         reducer( result );
         ^~~~~~~
Arborx/src/details/ArborX_DetailsUtils.hpp:179:9: note: (if you use ‘-fpermissive’, G++ will accept your code, but allowing the use of an undeclared name is deprecated)
Arborx/src/details/ArborX_DetailsUtils.hpp:182:55: error: ‘reducer’ was not declared in this scope
         Kokkos::Impl::min_max_functor<ViewType>( v ), reducer );
                                                       ^~~~~~~
```

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -124,5 +124,4 @@ RUN KOKKOS_VERSION=5d6e7fb38e96aec88d2c514e1f9be1cf2b549b57 && \
        --arch=SNB,Volta70 \
        --cxxstandard=c++14 && \
     make -j${NPROCS} install && \
-    rm -rf ${SCRATCH_DIR} && \
-    rm /root/kokkos.patch
+    rm -rf ${SCRATCH_DIR}

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -118,7 +118,7 @@ RUN KOKKOS_VERSION=5d6e7fb38e96aec88d2c514e1f9be1cf2b549b57 && \
     tar -xf ${KOKKOS_ARCHIVE} -C kokkos --strip-components=1 && \
     cd kokkos && \
     mkdir -p build && cd build && \
-    [ ! -z "${CUDA_VERSION}" ] && CUDA_OPTIONS="--with-cuda --with-cuda-options=enable_lambda,enable_uvm" || true && \
+    [ ! -z "${CUDA_VERSION}" ] && CUDA_OPTIONS="--with-cuda --with-cuda-options=enable_lambda" || true && \
     ../generate_makefile.bash --prefix=${KOKKOS_DIR} \
        --with-serial --with-openmp ${CUDA_OPTIONS} \
        --arch=SNB,Volta70 \

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -120,7 +120,8 @@ RUN KOKKOS_VERSION=5d6e7fb38e96aec88d2c514e1f9be1cf2b549b57 && \
     mkdir -p build && cd build && \
     [ ! -z "${CUDA_VERSION}" ] && CUDA_OPTIONS="--with-cuda --with-cuda-options=enable_lambda" || true && \
     ../generate_makefile.bash --prefix=${KOKKOS_DIR} \
-       --with-serial --with-openmp ${CUDA_OPTIONS} \
+       --with-serial --with-openmp \
+       --with-options=disable_deprecated_code ${CUDA_OPTIONS} \
        --arch=SNB,Volta70 \
        --cxxstandard=c++14 && \
     make -j${NPROCS} install && \

--- a/examples/bvh_driver/bvh_driver.cpp
+++ b/examples/bvh_driver/bvh_driver.cpp
@@ -163,8 +163,8 @@ void BM_knn_search(benchmark::State &state)
 
   for (auto _ : state)
   {
-    Kokkos::View<int *, DeviceType> offset("offset");
-    Kokkos::View<int *, DeviceType> indices("indices");
+    Kokkos::View<int *, DeviceType> offset("offset", 0);
+    Kokkos::View<int *, DeviceType> indices("indices", 0);
     auto const start = std::chrono::high_resolution_clock::now();
     index.query(queries, indices, offset);
     auto const end = std::chrono::high_resolution_clock::now();
@@ -194,8 +194,8 @@ void BM_radius_search(benchmark::State &state)
   bool first_pass = true;
   for (auto _ : state)
   {
-    Kokkos::View<int *, DeviceType> offset("offset");
-    Kokkos::View<int *, DeviceType> indices("indices");
+    Kokkos::View<int *, DeviceType> offset("offset", 0);
+    Kokkos::View<int *, DeviceType> indices("indices", 0);
     auto const start = std::chrono::high_resolution_clock::now();
     index.query(queries, indices, offset, buffer_size);
     auto const end = std::chrono::high_resolution_clock::now();

--- a/examples/distributed_tree_driver/distributed_tree_driver.cpp
+++ b/examples/distributed_tree_driver/distributed_tree_driver.cpp
@@ -193,7 +193,7 @@ int main_(std::vector<std::string> const &args)
   MPI_Comm_rank(comm, &comm_rank);
   int comm_size;
   MPI_Comm_size(comm, &comm_size);
-  Kokkos::View<ArborX::Point *, DeviceType> random_points("random_points");
+  Kokkos::View<ArborX::Point *, DeviceType> random_points("random_points", 0);
   {
     // Random points are "reused" between building the tree and performing
     // queries. Note that this means that for the points in the middle of
@@ -296,9 +296,9 @@ int main_(std::vector<std::string> const &args)
                          });
     Kokkos::fence();
 
-    Kokkos::View<int *, DeviceType> offset("offset");
-    Kokkos::View<int *, DeviceType> indices("indices");
-    Kokkos::View<int *, DeviceType> ranks("ranks");
+    Kokkos::View<int *, DeviceType> offset("offset", 0);
+    Kokkos::View<int *, DeviceType> indices("indices", 0);
+    Kokkos::View<int *, DeviceType> ranks("ranks", 0);
 
     auto knn = time_monitor.getNewTimer("knn");
     MPI_Barrier(comm);
@@ -328,9 +328,9 @@ int main_(std::vector<std::string> const &args)
                          });
     Kokkos::fence();
 
-    Kokkos::View<int *, DeviceType> offset("offset");
-    Kokkos::View<int *, DeviceType> indices("indices");
-    Kokkos::View<int *, DeviceType> ranks("ranks");
+    Kokkos::View<int *, DeviceType> offset("offset", 0);
+    Kokkos::View<int *, DeviceType> indices("indices", 0);
+    Kokkos::View<int *, DeviceType> ranks("ranks", 0);
 
     auto radius = time_monitor.getNewTimer("radius");
     MPI_Barrier(comm);

--- a/examples/viz/tree_visualization.cpp
+++ b/examples/viz/tree_visualization.cpp
@@ -36,7 +36,7 @@ void viz(std::string const &prefix, std::string const &infile, int n_neighbors)
 {
   using DeviceType = typename TreeType::device_type;
   using ExecutionSpace = typename DeviceType::execution_space;
-  Kokkos::View<ArborX::Point *, DeviceType> points("points");
+  Kokkos::View<ArborX::Point *, DeviceType> points("points", 0);
   loadPointCloud(infile, points);
 
   TreeType bvh(points);

--- a/src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp
+++ b/src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp
@@ -123,6 +123,27 @@ struct DistributedSearchTreeImpl
                     typename View::non_const_type imports);
 };
 
+namespace internal
+{
+template <typename PointerType>
+struct PointerDepth
+{
+  static int constexpr value = 0;
+};
+
+template <typename PointerType>
+struct PointerDepth<PointerType *>
+{
+  static int constexpr value = PointerDepth<PointerType>::value + 1;
+};
+
+template <typename PointerType, std::size_t N>
+struct PointerDepth<PointerType[N]>
+{
+  static int constexpr value = PointerDepth<PointerType>::value;
+};
+} // namespace internal
+
 template <typename View>
 inline Kokkos::View<typename View::traits::data_type, Kokkos::LayoutRight,
                     typename View::traits::host_mirror_space>
@@ -137,11 +158,18 @@ create_layout_right_mirror_view(
                      typename View::traits::host_mirror_space::memory_space>::
             value)>::type * = 0)
 {
+  constexpr int pointer_depth =
+      internal::PointerDepth<typename View::traits::data_type>::value;
   return Kokkos::View<typename View::traits::data_type, Kokkos::LayoutRight,
                       typename View::traits::host_mirror_space>(
       std::string(src.label()).append("_layout_right_mirror"), src.extent(0),
-      src.extent(1), src.extent(2), src.extent(3), src.extent(4), src.extent(5),
-      src.extent(6), src.extent(7));
+      pointer_depth > 1 ? src.extent(1) : KOKKOS_INVALID_INDEX,
+      pointer_depth > 2 ? src.extent(2) : KOKKOS_INVALID_INDEX,
+      pointer_depth > 3 ? src.extent(3) : KOKKOS_INVALID_INDEX,
+      pointer_depth > 4 ? src.extent(4) : KOKKOS_INVALID_INDEX,
+      pointer_depth > 5 ? src.extent(5) : KOKKOS_INVALID_INDEX,
+      pointer_depth > 6 ? src.extent(6) : KOKKOS_INVALID_INDEX,
+      pointer_depth > 7 ? src.extent(7) : KOKKOS_INVALID_INDEX);
 }
 
 template <typename View>

--- a/src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp
+++ b/src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp
@@ -452,8 +452,8 @@ void DistributedSearchTreeImpl<DeviceType>::sortResults(
   using Comp = Kokkos::BinOp1D<View>;
   using Value = typename View::non_const_value_type;
 
-  Kokkos::Experimental::MinMaxScalar<Value> result;
-  Kokkos::Experimental::MinMax<Value> reducer(result);
+  Kokkos::MinMaxScalar<Value> result;
+  Kokkos::MinMax<Value> reducer(result);
   parallel_reduce(Kokkos::RangePolicy<ExecutionSpace>(0, n),
                   Kokkos::Impl::min_max_functor<View>(keys), reducer);
   if (result.min_val == result.max_val)

--- a/src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp
+++ b/src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp
@@ -139,10 +139,9 @@ create_layout_right_mirror_view(
 {
   return Kokkos::View<typename View::traits::data_type, Kokkos::LayoutRight,
                       typename View::traits::host_mirror_space>(
-      std::string(src.label()).append("_layout_right_mirror"),
-      src.dimension_0(), src.dimension_1(), src.dimension_2(),
-      src.dimension_3(), src.dimension_4(), src.dimension_5(),
-      src.dimension_6(), src.dimension_7());
+      std::string(src.label()).append("_layout_right_mirror"), src.extent(0),
+      src.extent(1), src.extent(2), src.extent(3), src.extent(4), src.extent(5),
+      src.extent(6), src.extent(7));
 }
 
 template <typename View>
@@ -169,21 +168,20 @@ DistributedSearchTreeImpl<DeviceType>::sendAcrossNetwork(
     Distributor const &distributor, View exports,
     typename View::non_const_type imports)
 {
-  ARBORX_ASSERT(
-      (exports.dimension_0() == distributor.getTotalSendLength()) &&
-      (imports.dimension_0() == distributor.getTotalReceiveLength()) &&
-      (exports.dimension_1() == imports.dimension_1()) &&
-      (exports.dimension_2() == imports.dimension_2()) &&
-      (exports.dimension_3() == imports.dimension_3()) &&
-      (exports.dimension_4() == imports.dimension_4()) &&
-      (exports.dimension_5() == imports.dimension_5()) &&
-      (exports.dimension_6() == imports.dimension_6()) &&
-      (exports.dimension_7() == imports.dimension_7()));
+  ARBORX_ASSERT((exports.extent(0) == distributor.getTotalSendLength()) &&
+                (imports.extent(0) == distributor.getTotalReceiveLength()) &&
+                (exports.extent(1) == imports.extent(1)) &&
+                (exports.extent(2) == imports.extent(2)) &&
+                (exports.extent(3) == imports.extent(3)) &&
+                (exports.extent(4) == imports.extent(4)) &&
+                (exports.extent(5) == imports.extent(5)) &&
+                (exports.extent(6) == imports.extent(6)) &&
+                (exports.extent(7) == imports.extent(7)));
 
-  auto const num_packets = exports.dimension_1() * exports.dimension_2() *
-                           exports.dimension_3() * exports.dimension_4() *
-                           exports.dimension_5() * exports.dimension_6() *
-                           exports.dimension_7();
+  auto const num_packets = exports.extent(1) * exports.extent(2) *
+                           exports.extent(3) * exports.extent(4) *
+                           exports.extent(5) * exports.extent(6) *
+                           exports.extent(7);
 
   auto exports_host = create_layout_right_mirror_view(exports);
   Kokkos::deep_copy(exports_host, exports);

--- a/src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp
+++ b/src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp
@@ -313,7 +313,7 @@ void DistributedSearchTreeImpl<DeviceType>::queryDispatch(
   auto const &bottom_tree = tree._bottom_tree;
   auto comm = tree._comm;
 
-  Kokkos::View<double *, DeviceType> distances("distances");
+  Kokkos::View<double *, DeviceType> distances("distances", 0);
   if (distances_ptr)
     distances = *distances_ptr;
 
@@ -345,8 +345,8 @@ void DistributedSearchTreeImpl<DeviceType>::queryDispatch(
     ////////////////////////////////////////////////////////////////////////////
     // Forward queries
     ////////////////////////////////////////////////////////////////////////////
-    Kokkos::View<int *, DeviceType> ids("query_ids");
-    Kokkos::View<Query *, DeviceType> fwd_queries("fwd_queries");
+    Kokkos::View<int *, DeviceType> ids("query_ids", 0);
+    Kokkos::View<Query *, DeviceType> fwd_queries("fwd_queries", 0);
     forwardQueries(comm, queries, indices, offset, fwd_queries, ids, ranks);
     ////////////////////////////////////////////////////////////////////////////
 
@@ -394,8 +394,8 @@ void DistributedSearchTreeImpl<DeviceType>::queryDispatch(
   ////////////////////////////////////////////////////////////////////////////
   // Forward queries
   ////////////////////////////////////////////////////////////////////////////
-  Kokkos::View<int *, DeviceType> ids("query_ids");
-  Kokkos::View<Query *, DeviceType> fwd_queries("fwd_queries");
+  Kokkos::View<int *, DeviceType> ids("query_ids", 0);
+  Kokkos::View<Query *, DeviceType> fwd_queries("fwd_queries", 0);
   forwardQueries(comm, queries, indices, offset, fwd_queries, ids, ranks);
   ////////////////////////////////////////////////////////////////////////////
 

--- a/src/details/ArborX_DetailsSortUtils.hpp
+++ b/src/details/ArborX_DetailsSortUtils.hpp
@@ -37,8 +37,8 @@ sortObjects(Kokkos::View<unsigned int *, DeviceType> view)
   using CompType = Kokkos::BinOp1D<ViewType>;
   using ExecutionSpace = typename DeviceType::execution_space;
 
-  Kokkos::Experimental::MinMaxScalar<ValueType> result;
-  Kokkos::Experimental::MinMax<ValueType> reducer(result);
+  Kokkos::MinMaxScalar<ValueType> result;
+  Kokkos::MinMax<ValueType> reducer(result);
   parallel_reduce(ARBORX_MARK_REGION("find_min_max_view"),
                   Kokkos::RangePolicy<ExecutionSpace>(0, n),
                   Kokkos::Impl::min_max_functor<ViewType>(view), reducer);

--- a/src/details/ArborX_DetailsUtils.hpp
+++ b/src/details/ArborX_DetailsUtils.hpp
@@ -120,7 +120,7 @@ lastElement(Kokkos::View<T, P...> const &v)
   auto v_subview = Kokkos::subview(v, n - 1);
   auto v_host = Kokkos::create_mirror_view(v_subview);
   Kokkos::deep_copy(v_host, v_subview);
-  return v_host(0);
+  return v_host();
 }
 
 /** \brief Fills the view with a sequence of numbers

--- a/src/details/ArborX_DetailsUtils.hpp
+++ b/src/details/ArborX_DetailsUtils.hpp
@@ -169,10 +169,8 @@ minMax(ViewType const &v)
   using ExecutionSpace = typename ViewType::execution_space;
   auto const n = v.extent(0);
   ARBORX_ASSERT(n > 0);
-  Kokkos::Experimental::MinMaxScalar<typename ViewType::non_const_value_type>
-      result;
-  Kokkos::Experimental::MinMax<typename ViewType::non_const_value_type> reducer(
-      result);
+  Kokkos::MinMaxScalar<typename ViewType::non_const_value_type> result;
+  Kokkos::MinMax<typename ViewType::non_const_value_type> reducer(result);
   Kokkos::parallel_reduce("minMax", Kokkos::RangePolicy<ExecutionSpace>(0, n),
                           Kokkos::Impl::min_max_functor<ViewType>(v), reducer);
   return std::make_pair(result.min_val, result.max_val);
@@ -190,8 +188,7 @@ typename ViewType::non_const_value_type min(ViewType const &v)
   auto const n = v.extent(0);
   ARBORX_ASSERT(n > 0);
   typename ViewType::non_const_value_type result;
-  Kokkos::Experimental::Min<typename ViewType::non_const_value_type> reducer(
-      result);
+  Kokkos::Min<typename ViewType::non_const_value_type> reducer(result);
   Kokkos::parallel_reduce(Kokkos::RangePolicy<ExecutionSpace>(0, n),
                           KOKKOS_LAMBDA(int i, int &update) {
                             if (v(i) < update)
@@ -213,8 +210,7 @@ typename ViewType::non_const_value_type max(ViewType const &v)
   auto const n = v.extent(0);
   ARBORX_ASSERT(n > 0);
   typename ViewType::non_const_value_type result;
-  Kokkos::Experimental::Max<typename ViewType::non_const_value_type> reducer(
-      result);
+  Kokkos::Max<typename ViewType::non_const_value_type> reducer(result);
   Kokkos::parallel_reduce(Kokkos::RangePolicy<ExecutionSpace>(0, n),
                           KOKKOS_LAMBDA(int i, int &update) {
                             if (v(i) > update)

--- a/test/Search_UnitTestHelpers.hpp
+++ b/test/Search_UnitTestHelpers.hpp
@@ -32,8 +32,8 @@ void checkResults(ArborX::BVH<DeviceType> const &bvh,
                   std::vector<int> const &indices_ref,
                   std::vector<int> const &offset_ref)
 {
-  Kokkos::View<int *, DeviceType> indices("indices");
-  Kokkos::View<int *, DeviceType> offset("offset");
+  Kokkos::View<int *, DeviceType> indices("indices", 0);
+  Kokkos::View<int *, DeviceType> offset("offset", 0);
   bvh.query(queries, indices, offset);
 
   auto indices_host = Kokkos::create_mirror_view(indices);
@@ -55,9 +55,9 @@ void checkResults(ArborX::BVH<DeviceType> const &bvh,
                   std::vector<int> const &offset_ref,
                   std::vector<double> const &distances_ref)
 {
-  Kokkos::View<int *, DeviceType> indices("indices");
-  Kokkos::View<int *, DeviceType> offset("offset");
-  Kokkos::View<double *, DeviceType> distances("distances");
+  Kokkos::View<int *, DeviceType> indices("indices", 0);
+  Kokkos::View<int *, DeviceType> offset("offset", 0);
+  Kokkos::View<double *, DeviceType> distances("distances", 0);
   bvh.query(queries, indices, offset, distances);
 
   auto indices_host = Kokkos::create_mirror_view(indices);
@@ -79,9 +79,9 @@ void checkResults(ArborX::DistributedSearchTree<DeviceType> const &tree,
                   std::vector<int> const &offset_ref,
                   std::vector<int> const &ranks_ref)
 {
-  Kokkos::View<int *, DeviceType> indices("indices");
-  Kokkos::View<int *, DeviceType> offset("offset");
-  Kokkos::View<int *, DeviceType> ranks("ranks");
+  Kokkos::View<int *, DeviceType> indices("indices", 0);
+  Kokkos::View<int *, DeviceType> offset("offset", 0);
+  Kokkos::View<int *, DeviceType> ranks("ranks", 0);
   tree.query(queries, indices, offset, ranks);
 
   auto indices_host = Kokkos::create_mirror_view(indices);
@@ -124,10 +124,10 @@ void checkResults(ArborX::DistributedSearchTree<DeviceType> const &tree,
                   std::vector<int> const &ranks_ref,
                   std::vector<double> const &distances_ref)
 {
-  Kokkos::View<int *, DeviceType> indices("indices");
-  Kokkos::View<int *, DeviceType> offset("offset");
-  Kokkos::View<int *, DeviceType> ranks("ranks");
-  Kokkos::View<double *, DeviceType> distances("distances");
+  Kokkos::View<int *, DeviceType> indices("indices", 0);
+  Kokkos::View<int *, DeviceType> offset("offset", 0);
+  Kokkos::View<int *, DeviceType> ranks("ranks", 0);
+  Kokkos::View<double *, DeviceType> distances("distances", 0);
   tree.query(queries, indices, offset, ranks, distances);
 
   auto indices_host = Kokkos::create_mirror_view(indices);

--- a/test/tstDetailsBatchedQueries.cpp
+++ b/test/tstDetailsBatchedQueries.cpp
@@ -33,10 +33,10 @@ Kokkos::View<ValueType *, DeviceType> toView(std::vector<ValueType> const &v)
 BOOST_AUTO_TEST_CASE_TEMPLATE(permute_offset_and_indices, DeviceType,
                               ARBORX_DEVICE_TYPES)
 {
-  Kokkos::View<int *, DeviceType> offset("offset");
-  Kokkos::View<int *, DeviceType> indices("indices");
+  Kokkos::View<int *, DeviceType> offset("offset", 0);
+  Kokkos::View<int *, DeviceType> indices("indices", 0);
 
-  Kokkos::View<size_t *, DeviceType> permute("permute");
+  Kokkos::View<size_t *, DeviceType> permute("permute", 0);
 
   BOOST_CHECK_THROW(
       ArborX::Details::BatchedQueries<DeviceType>::reversePermutation(

--- a/test/tstDetailsDistributedSearchTreeImpl.cpp
+++ b/test/tstDetailsDistributedSearchTreeImpl.cpp
@@ -132,14 +132,14 @@ inline void checkViewWasNotAllocated(View1 const &v1, View2 const &v2)
   BOOST_TEST((std::is_same<typename View1::memory_space,
                            typename View2::memory_space>::value));
 
-  BOOST_TEST(v1.dimension_0() == v2.dimension_0());
-  BOOST_TEST(v1.dimension_1() == v2.dimension_1());
-  BOOST_TEST(v1.dimension_2() == v2.dimension_2());
-  BOOST_TEST(v1.dimension_3() == v2.dimension_3());
-  BOOST_TEST(v1.dimension_4() == v2.dimension_4());
-  BOOST_TEST(v1.dimension_5() == v2.dimension_5());
-  BOOST_TEST(v1.dimension_6() == v2.dimension_6());
-  BOOST_TEST(v1.dimension_7() == v2.dimension_7());
+  BOOST_TEST(v1.extent(0) == v2.extent(0));
+  BOOST_TEST(v1.extent(1) == v2.extent(1));
+  BOOST_TEST(v1.extent(2) == v2.extent(2));
+  BOOST_TEST(v1.extent(3) == v2.extent(3));
+  BOOST_TEST(v1.extent(4) == v2.extent(4));
+  BOOST_TEST(v1.extent(5) == v2.extent(5));
+  BOOST_TEST(v1.extent(6) == v2.extent(6));
+  BOOST_TEST(v1.extent(7) == v2.extent(7));
 }
 
 template <typename View1, typename View2>
@@ -151,14 +151,14 @@ inline void checkNewViewWasAllocated(View1 const &v1, View2 const &v2)
   BOOST_TEST((std::is_same<typename View1::const_value_type,
                            typename View2::const_value_type>::value));
 
-  BOOST_TEST(v1.dimension_0() == v2.dimension_0());
-  BOOST_TEST(v1.dimension_1() == v2.dimension_1());
-  BOOST_TEST(v1.dimension_2() == v2.dimension_2());
-  BOOST_TEST(v1.dimension_3() == v2.dimension_3());
-  BOOST_TEST(v1.dimension_4() == v2.dimension_4());
-  BOOST_TEST(v1.dimension_5() == v2.dimension_5());
-  BOOST_TEST(v1.dimension_6() == v2.dimension_6());
-  BOOST_TEST(v1.dimension_7() == v2.dimension_7());
+  BOOST_TEST(v1.extent(0) == v2.extent(0));
+  BOOST_TEST(v1.extent(1) == v2.extent(1));
+  BOOST_TEST(v1.extent(2) == v2.extent(2));
+  BOOST_TEST(v1.extent(3) == v2.extent(3));
+  BOOST_TEST(v1.extent(4) == v2.extent(4));
+  BOOST_TEST(v1.extent(5) == v2.extent(5));
+  BOOST_TEST(v1.extent(6) == v2.extent(6));
+  BOOST_TEST(v1.extent(7) == v2.extent(7));
 }
 
 BOOST_AUTO_TEST_CASE_TEMPLATE(create_layout_right_mirror_view, DeviceType,

--- a/test/tstDetailsDistributedSearchTreeImpl.cpp
+++ b/test/tstDetailsDistributedSearchTreeImpl.cpp
@@ -108,7 +108,7 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(count_results, DeviceType, ARBORX_DEVICE_TYPES)
     ids_host(i) = ids_ref[i];
   Kokkos::deep_copy(ids, ids_host);
 
-  Kokkos::View<int *, DeviceType> offset("offset");
+  Kokkos::View<int *, DeviceType> offset("offset", m);
 
   ArborX::Details::DistributedSearchTreeImpl<DeviceType>::countResults(m, ids,
                                                                        offset);

--- a/test/tstDetailsDistributedSearchTreeImpl.cpp
+++ b/test/tstDetailsDistributedSearchTreeImpl.cpp
@@ -237,3 +237,32 @@ BOOST_AUTO_TEST_CASE(sort_and_determine_buffer_layout)
   checkBufferLayout({0, 1, 2, 3}, {3, 2, 1, 0}, {3, 2, 1, 0}, {1, 1, 1, 1},
                     {0, 1, 2, 3, 4});
 }
+
+BOOST_AUTO_TEST_CASE(pointer_depth)
+{
+  static_assert(ArborX::Details::internal::PointerDepth<double>::value == 0,
+                "Failing for double");
+  static_assert(ArborX::Details::internal::PointerDepth<double *>::value == 1,
+                "Failing for double*");
+  static_assert(ArborX::Details::internal::PointerDepth<double[2]>::value == 0,
+                "Failing for double[2]");
+  static_assert(ArborX::Details::internal::PointerDepth<double **>::value == 2,
+                "Failing for double**");
+  static_assert(ArborX::Details::internal::PointerDepth<double * [2]>::value ==
+                    1,
+                "Failing for double*[2]");
+  static_assert(ArborX::Details::internal::PointerDepth<double[2][3]>::value ==
+                    0,
+                "Failing for double[2][3]");
+  static_assert(ArborX::Details::internal::PointerDepth<double ***>::value == 3,
+                "Failing for double***");
+  static_assert(
+      ArborX::Details::internal::PointerDepth<double * * [2]>::value == 2,
+      "Failing for double[2]");
+  static_assert(
+      ArborX::Details::internal::PointerDepth<double * [2][3]>::value == 1,
+      "Failing for double*[2][3]");
+  static_assert(
+      ArborX::Details::internal::PointerDepth<double[2][3][4]>::value == 0,
+      "Failing for double[2][3][4]");
+}

--- a/test/tstDetailsDistributedSearchTreeImpl.cpp
+++ b/test/tstDetailsDistributedSearchTreeImpl.cpp
@@ -186,10 +186,20 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(create_layout_right_mirror_view, DeviceType,
   auto v_h = create_layout_right_mirror_view(v);
   checkViewWasNotAllocated(v, v_h);
 
+  // the same with compile time size
+  View<int[2][3], LayoutRight, DeviceType> v_c("v");
+  auto v_c_h = create_layout_right_mirror_view(v_c);
+  checkViewWasNotAllocated(v_c, v_c_h);
+
   // left layout and rank > 1 -> allocate
   View<int **, LayoutLeft, DeviceType> w("w", 4, 5);
   auto w_h = create_layout_right_mirror_view(w);
   checkNewViewWasAllocated(w, w_h);
+
+  // the same with compile time size
+  View<int * [5], LayoutLeft, DeviceType> w_c("v", 4);
+  auto w_c_h = create_layout_right_mirror_view(w_c);
+  checkNewViewWasAllocated(w_c, w_c_h);
 
   // strided layout -> allocate
   auto x = subview(v, ALL, 0);

--- a/test/tstDetailsUtils.cpp
+++ b/test/tstDetailsUtils.cpp
@@ -125,7 +125,7 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(minmax, DeviceType, ARBORX_DEVICE_TYPES)
   auto const result_float = ArborX::minMax(v);
   BOOST_TEST(std::get<0>(result_float) == 1.41);
   BOOST_TEST(std::get<1>(result_float) == 3.14);
-  Kokkos::View<int *, DeviceType> w("w");
+  Kokkos::View<int *, DeviceType> w("w", 0);
   BOOST_CHECK_THROW(ArborX::minMax(w), ArborX::SearchException);
   Kokkos::resize(w, 1);
   Kokkos::deep_copy(w, 255);

--- a/test/tstDetailsUtils.cpp
+++ b/test/tstDetailsUtils.cpp
@@ -107,7 +107,7 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(last_element, DeviceType, ARBORX_DEVICE_TYPES)
   BOOST_TEST(ArborX::lastElement(v) == 24);
   Kokkos::View<int *, DeviceType> w("w", 0);
   BOOST_CHECK_THROW(ArborX::lastElement(w), ArborX::SearchException);
-  Kokkos::View<double[1], DeviceType> u("u", 1);
+  Kokkos::View<double[1], DeviceType> u("u");
   Kokkos::deep_copy(u, 3.14);
   BOOST_TEST(ArborX::lastElement(u) == 3.14);
 }

--- a/test/tstDistributedSearchTree.cpp
+++ b/test/tstDistributedSearchTree.cpp
@@ -433,9 +433,9 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(boost_comparison, DeviceType, ARBORX_DEVICE_TYPES)
   Kokkos::fence();
 
   // Perform the search
-  Kokkos::View<int *, DeviceType> indices("indices");
-  Kokkos::View<int *, DeviceType> offset("offset");
-  Kokkos::View<int *, DeviceType> ranks("ranks");
+  Kokkos::View<int *, DeviceType> indices("indices", 0);
+  Kokkos::View<int *, DeviceType> offset("offset", 0);
+  Kokkos::View<int *, DeviceType> ranks("ranks", 0);
   distributed_tree.query(within_queries, indices, offset, ranks);
   auto indices_host = Kokkos::create_mirror_view(indices);
   Kokkos::deep_copy(indices_host, indices);

--- a/test/tstLinearBVH.cpp
+++ b/test/tstLinearBVH.cpp
@@ -248,8 +248,8 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(buffer_optimization, DeviceType,
   });
 
   using ViewType = Kokkos::View<int *, DeviceType>;
-  ViewType indices("indices");
-  ViewType offset("offset");
+  ViewType indices("indices", 0);
+  ViewType offset("offset", 0);
 
   std::vector<int> indices_ref = {3, 2, 1, 0};
   std::vector<int> offset_ref = {0, 0, 4, 4};
@@ -311,8 +311,8 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(not_exceeding_stack_capacity, DeviceType,
   }
   auto const bvh = makeBvh<DeviceType>(boxes);
 
-  Kokkos::View<int *, DeviceType> indices("indices");
-  Kokkos::View<int *, DeviceType> offset("offset");
+  Kokkos::View<int *, DeviceType> indices("indices", 0);
+  Kokkos::View<int *, DeviceType> offset("offset", 0);
   // query number of nearest neighbors that exceed capacity of the stack is
   // not a problem
   BOOST_CHECK_NO_THROW(bvh.query(makeNearestQueries<DeviceType>({
@@ -345,7 +345,7 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(miscellaneous, DeviceType, ARBORX_DEVICE_TYPES)
   Kokkos::View<int *, DeviceType> zeros("zeros", 3);
   Kokkos::deep_copy(zeros, 255);
   Kokkos::View<Kokkos::pair<int, double> *, DeviceType> empty_buffer(
-      "empty_buffer");
+      "empty_buffer", 0);
   Kokkos::parallel_for(
       Kokkos::RangePolicy<ExecutionSpace>(0, 1), KOKKOS_LAMBDA(int) {
         ArborX::Point p = {{0., 0., 0.}};
@@ -742,8 +742,8 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(boost_rtree, DeviceType, ARBORX_DEVICE_TYPES)
   auto rtree_results =
       BoostRTreeHelpers::performQueries(rtree, nearest_queries_host);
 
-  Kokkos::View<int *, DeviceType> offset_nearest("offset_nearest");
-  Kokkos::View<int *, DeviceType> indices_nearest("indices_nearest");
+  Kokkos::View<int *, DeviceType> offset_nearest("offset_nearest", 0);
+  Kokkos::View<int *, DeviceType> indices_nearest("indices_nearest", 0);
   bvh.query(nearest_queries, indices_nearest, offset_nearest);
   auto offset_nearest_host = Kokkos::create_mirror_view(offset_nearest);
   Kokkos::deep_copy(offset_nearest_host, offset_nearest);
@@ -762,8 +762,8 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(boost_rtree, DeviceType, ARBORX_DEVICE_TYPES)
   bvh_results = std::make_tuple(offset_nearest_host, indices_nearest_host);
   validateResults(rtree_results, bvh_results);
 
-  Kokkos::View<int *, DeviceType> offset_within("offset_within");
-  Kokkos::View<int *, DeviceType> indices_within("indices_within");
+  Kokkos::View<int *, DeviceType> offset_within("offset_within", 0);
+  Kokkos::View<int *, DeviceType> indices_within("indices_within", 0);
   bvh.query(within_queries, indices_within, offset_within);
   auto offset_within_host = Kokkos::create_mirror_view(offset_within);
   Kokkos::deep_copy(offset_within_host, offset_within);

